Docs: update plugins tutorials (#373)
grafana/plugin-tools: docusaurus/docs/shared/plugin-anatomy.md

> Plugins come in different shapes and sizes, however every plugin you create will require at least two files: `plugin.json` and `src/module.ts`.

Reviewer: ```suggestion
Every plugin you create requires at least two files: `plugin.json` and `src/module.ts`.
```

> ### plugin.json

Reviewer: ```suggestion
### `plugin.json`
```

Reviewer 2: happy to make this change, but shouldnt we also make it for module.ts?

> When Grafana starts, it scans the [plugin directory](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#plugins) for any subdirectory that contains a `plugin.json` file. The `plugin.json` file contains information about your plugin, and tells Grafana about what capabilities and dependencies your plugin needs.

Reviewer: ```suggestion
When Grafana starts, it scans the [plugin directory](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#plugins) for any subdirectory that contains a `plugin.json` file. The `plugin.json` file contains information about your plugin and tells Grafana about what capabilities and dependencies your plugin needs.
```

While certain plugin types can have specific configuration options, let's look at the mandatory ones:

- `type` tells Grafana what type of plugin to expect. Grafana supports three types of plugins: `panel`, `datasource`, and `app`.
- `name` is what users will see in the list of plugins. If you're creating a data source, this is typically the name of the database it connects to, such as Prometheus, PostgreSQL, or Stackdriver.

> - `id` uniquely identifies your plugin and should follow the Grafana naming convention - `<$organization-name>-<$plugin-name>-<$plugin-type>`. The create-plugin tool correctly configures this based on your responses to its prompts.

Reviewer: ```suggestion
- `id` uniquely identifies your plugin and should follow this naming convention:`<$organization-name>-<$plugin-name>-<$plugin-type>`. The create-plugin tool correctly configures this based on your responses to its prompts.
```
